OpenRA.Game/TranslationProvider.cs

```diff
@@ -0,0 +1,62 @@
+#region Copyright & License Information
+/*
+ * Copyright (c) The OpenRA Developers and Contributors
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation, either version 3 of
+ * the License, or (at your option) any later version. For more
+ * information, see COPYING.
+ */
+#endregion
+
+using System.Collections.Generic;
+using OpenRA.FileSystem;
+
+namespace OpenRA
+{
+	public static class TranslationProvider
+	{
+		static Translation modTranslation;
+		static Translation mapTranslation;
+
+		public static void Initialize(ModData modData, IReadOnlyFileSystem fileSystem)
```

Comment:
Can we put in a `lock` around all the static methods please? For thread-safety.

Reply:
What exactly do we want to lock?

Reply:
Wrapping the entire method bodies should be fine. Introducing a dedicated `static readonly object SyncObject = new object();` to lock on would suffice.

Reply:
<img width="1103" alt="Screenshot 2023-04-21 at 23 47 01" src="https://user-images.githubusercontent.com/37534529/233731859-3f2c53ed-e30e-422b-95a8-40589c8f3293.png">

do you mean something like this?

Reply:
Just one object like in the code example needs to be locked.

Reply:
I don't know if this is really necessary though, three calls are only reading and initialize shouldn't be called from different threads or is it? Do you have a specific case in mind @RoosterDragon, like a new translation (map) being initialized while something else is reading?

Reply:
For static methods, the callers have no way to co-ordinate different threads - anybody can call static methods from anywhere at any time. Static methods are therefore always on the hook to ensure they provide thread-safety.

We might not intend to ever have these called concurrently, but if they ever are then then lock ensures we don't get weird, "impossible" seeming crashes. Whilst it's not common, we do use threads is a few places - for example we load replays on a background thread currently. If that would trigger translations (either now or in the future) then we need to avoid crashing horribly if some other thread is mutating the shared translation state.

```c#
static readonly object SyncObject = new object(); // or whatever name you prefer.
static Translation modTranslation;
static Translation mapTranslation;

public static void Initialize(...)
{
    lock (SyncObject)
    {
        // method body here
    }
}

//repeat for other methods
```

Reply:
Added